### PR TITLE
WT-10078 Release pages during stats tree walk and enable explicit statistics config for test/format

### PR DESCRIPTION
**Pull request by mickgraham-mongodb**
Fixing issue with bt_stat not releasing pages during walks. Updating test/format to have explicit settings for statistics mode and statistics_log sources setting.

### CODE REVIEW

File: src/btree/bt_stat.c [hunk 1]
```diff
@@ -86,11 +86,21 @@ __stat_tree_walk(WT_SESSION_IMPL *session)
...
-    while (
-      (ret = __wt_tree_walk(session, &next_walk, WT_READ_VISIBLE_ALL)) == 0 && next_walk != NULL) {
+
+    /*
+     * Pages read for statistics aren't "useful"; don't update the read generation of pages already
+     * in memory, and if a page is read, set its generation to a low value so it is evicted quickly.
+     * Same as with compact.
+     */
+    while ((ret = __wt_tree_walk(session, &next_walk,
+              WT_READ_NO_GEN | WT_READ_WONT_NEED | WT_READ_VISIBLE_ALL)) == 0 &&
```
Comment (sueloverso) [inline src/btree/bt_stat.c:96]: Nit: Alphabetize the flags please. Also fix the alphabetizing of the same ones in bt_compact.c and txn_rollback_to_stable.c.

File: test/format/config.sh [hunk 2]
```diff
@@ -242,6 +242,10 @@ CONFIG configuration_list[] = {
 
 {"runs.verify_failure_dump", "configure page dump on repeatable read error", C_BOOL | C_IGNORE, 0, 0, 1}
 
+{"statistics.mode", "statistics mode (fast | all)", C_IGNORE | C_STRING, 0, 0, 0}
```
Comment (sueloverso) [inline test/format/config.sh:245]: Alphabetize (all | fast ).

**Review by sueloverso — APPROVED** [review]
LGTM with a couple minor nits. I think formalizing the stats sources is better than the random line in wts.c

File: src/btree/bt_stat.c (hunk 1 above)
Comment (mickgraham-mongodb) [reply src/btree/bt_stat.c]: Done.

File: test/format/config.sh (hunk 2 above)
Comment (mickgraham-mongodb) [reply test/format/config.sh]: Done, and fixed off | file: as well

**Review by agorrod — APPROVED** [review]
LGTM

**Pull request merged**